soufi/tests/functional/test_functional.py

```diff
@@ -21,8 +22,12 @@
 
 
 class FunctionalFinderTests(testtools.TestCase):
+    # NOTE(nic): For test runners like PyCharm, set the environment variable
+    #  `RUN_FUNCTIONAL_TESTS=0` in the main test configuration to skip these
+    #  tests.  tox/stestr should already ignore them unless specifically asked.
     def setUp(self):
-        self.skipTest("functional tests disabled for unit test runs")
+        if not int(os.environ.get('RUN_FUNCTIONAL_TESTS', 1)):
+            self.skipTest("functional tests disabled for unit test runs")
```

Comment:
I think a better way of doing this is to move the `functional` dir up a level to `soufi/functional` and have two separate trees of tests. Then you don't need all the hacky environment variable & stestr exclusions.

Reply:
Eww.  But OK.